pkg/k8s/watchers/cilium_endpoint_slice.go

```diff
@@ -58,7 +82,13 @@ func (k *K8sWatcher) ciliumEndpointSliceInit(client *k8s.K8sCiliumClient, asyncC
 				},
 			},
 			nil,
+			cache.Indexers{
+				"localNode": CreateCiliumEndpointSliceLocalPodIndexFunc(),
+			},
 		)
+		k.ciliumEndpointSliceIndexerMU.Lock()
+		k.ciliumEndpointSliceIndexer = cesIndexer
+		k.ciliumEndpointSliceIndexerMU.Unlock()
```

Comment:
Why don't we use the `k.SetIndexer` function here?

Reply:
consistency, its done like that for every resource type: https://github.com/cilium/cilium/blob/master/pkg/k8s/watchers/pod.go#L129. SetStore had a note it was only for testing, but it didn't appear to be used at all so I removed it.